rules/java-net/src/main/resources/otagent/java-net.btm

```diff
@@ -0,0 +1,110 @@
+
+RULE java-net: Java HttpUrlConnection connect entry
+CLASS ^java.net.HttpURLConnection
+METHOD connect()
+HELPER io.opentracing.contrib.agent.OpenTracingHelper
+AT ENTRY
+IF !$0.connected && retrieveSpan($0) == null && !ignore($0)
+DO
+  associateSpan($0, getTracer().buildSpan($0.getRequestMethod())
```

Comment:
@yurishkuro Yes this is definitely an area for improvement - but it affects all of the http framework integrations, so I think it would be good to provide another opentracing-contrib project to tackle this (e.g. with annotations) and then the agent just hook into that.